backend/order.js

```diff
@@ -94,15 +102,20 @@ var order = {
       data,
       function(err, res) {
         if (err) {
-          throw err;
+          log.error('Failed to update contact details', {error: err});
+          return cb({err: true});
         }
+
         // TODO how we should really propagate these errors
         if (res.changedRows !== 1) {
-          cb({
+          errmsg = 'Should have updated one row, updated really ' + res.changedRows + ' rows';
+          log.error(errmsg);
+          return cb({
             err: true,
-            errmsg: 'Should have updated one row, updated really ' + res.changedRows + ' rows'
+            errmsg: errmsg
```

Comment:
Is there some reason why here we callback also the message instead of just err: true?